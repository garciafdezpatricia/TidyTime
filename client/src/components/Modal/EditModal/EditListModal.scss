@use "../../../../styles/variables" as *;

.edit-modal {
    display: flex;
    flex-direction: column;
    background-color: $white;
    border-radius: 5px;
    position: absolute;
    top: 10%;
    left: 50%;
    transform: translateX(-50%);
    padding: .5rem .5rem 1rem;
    border: solid 1px $primary-grey;
    width: 20rem;
    justify-content: center;
    align-items: center;
    cursor: default;
    font-size: .9rem;
    box-shadow: 0px 14px 42px -6px rgba(0,0,0,0.75);
    button {
        display: flex;
        justify-content: center;
        gap: .4rem;
    }

    @media screen and (min-width: 320px) and (max-width: 768px) {
        width: 18rem;
    }
}

body.dark-mode .edit-modal {
    color: $background-dark-0;
}

.edit-modal-header {
    display: flex;
    width: 100%;
    justify-content: flex-end;
    button {
        background-color: transparent;
        border: none;
        outline: none;
        color: $light-grey;
        &:hover {
            color: $black;
        }
    }
}

.rename-section {
    display: flex;
    gap: .5rem;
    margin-top: 1rem;
    margin-bottom: .5rem;
}

.input-section {
    display: flex;
    gap: .5rem;

    input {
        border-radius: .4rem;
        outline: none;
        border: .1rem solid $light-grey;
        padding-left: .1rem;
        width: 9rem;
    }
}

.confirm-rename {
    padding-inline: .4rem;
    background-color: $white;
    outline: none;
    border: .1rem solid $primary-green;
    color: $primary-green;
    &:not(:disabled):hover {
        background-color: $primary-green;
        color: $white;
    }
    &:disabled {
        border: .1rem solid $light-grey;
        color: $light-grey;
        cursor: auto;
    }
}

.rename-button {
    width: fit-content;
    padding-inline: .4rem;
    border: .13rem solid $dark-grey;
    background-color: $white;
    &:hover {
        background-color: $primary-grey;
    }
}

.remove-button {
    width: fit-content;
    padding-inline: .4rem;
    border: .13rem solid $remove-red;
    background-color: $white;
    &:hover {
        background-color: $hover-red;
    }
}