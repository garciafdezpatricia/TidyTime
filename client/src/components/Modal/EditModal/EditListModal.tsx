import { useEffect, useRef, useState } from "react";
import { MdOutlineDriveFileRenameOutline } from "react-icons/md";
import { RiDeleteBin5Fill } from "react-icons/ri";
import { FaCheck } from "react-icons/fa";
import { ImCross } from "react-icons/im";

export interface Props {
    title?: string
    onRenameAction: (arg?:any) => void | any;
    onDeleteAction: (arg?:any) => void | any;
    onInputChange: (arg?:any) => void | any;
    onClose: (arg?:any) => void | any;
    dontShowDelete?: boolean
}

export default function EditListModal({onRenameAction, onDeleteAction, onInputChange, onClose, dontShowDelete, title} : Props) {

    const [showInput, setShowInput] = useState(false);
    const [isButtonDisabled, setDisabled] = useState(false);
    const [newName, setNewName] = useState("")
    const nameInput = useRef(null);

    const handleKeyDown = (e:any) => {
        if (e.key === 'Enter' && showInput && newName !== "") {
            handleRename();
        }
    };

    const handleRename = () => {
        onRenameAction();
        // @ts-ignore
        nameInput.current.value = "";
        setNewName("");
    }

    const handleInputChange = (e:any) => {
        setNewName(e.target.value);
        onInputChange(e.target.value)
    }

    useEffect(() => {
        newName === ""
        ? setDisabled(true)
        : setDisabled(false);
    }, [newName])

    return (
        <>
            <div className="backdrop">
                <div className="edit-modal">
                <header className="edit-modal-header">
                    <button 
                        title="Close"
                        onClick={onClose} 
                        className="close-button">
                            <ImCross size={".7rem"} />
                    </button>
                </header>
                    <p>{title}</p>
                    <section className="rename-section">
                        <button 
                            onClick={() => setShowInput(!showInput)} 
                            className="rename-button"
                        >
                            <MdOutlineDriveFileRenameOutline />
                            Rename
                        </button>
                        {
                            showInput && (
                                <section className="input-section">
                                    <input 
                                        ref={nameInput}
                                        type="text" 
                                        onChange={(e) => handleInputChange(e)}
                                        onKeyDown={handleKeyDown}
                                    />
                                    <button className="confirm-rename" onClick={handleRename} disabled={isButtonDisabled}>
                                        <FaCheck />
                                    </button>
                                </section>
                            )
                        }
                    </section>
                    {
                        !dontShowDelete && 
                        <button 
                        onClick={onDeleteAction} 
                        className="remove-button"
                    >
                        <RiDeleteBin5Fill color="#b13838"/>
                        Remove
                    </button>
                    }
                </div>
            </div>
        </>
    )
}