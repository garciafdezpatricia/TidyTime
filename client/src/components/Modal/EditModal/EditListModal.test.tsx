import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import EditListModal from '@/src/components/Modal/EditModal/EditListModal';

describe('EditListModal', () => {
    it('renders without crashing', () => {
        const { container } = render(
        <EditListModal 
            onRenameAction={() => {}} 
            onDeleteAction={() => {}} 
            onInputChange={() => {}} 
            onClose={() => {}}      
            />);
        expect(container).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(
        <EditListModal 
            onRenameAction={() => {}} 
            onDeleteAction={() => {}} 
            onInputChange={() => {}} 
            onClose={() => {}}      
            />);
        expect(asFragment()).toMatchSnapshot();
      });
  });