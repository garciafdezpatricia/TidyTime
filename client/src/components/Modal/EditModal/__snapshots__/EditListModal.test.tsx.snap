// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditListModal matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="backdrop"
  >
    <div
      class="edit-modal"
      data-testid="edit-modal"
    >
      <header
        class="edit-modal-header"
      >
        <button
          class="close-button"
        >
          <svg
            fill="currentColor"
            height=".7rem"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 16 16"
            width=".7rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.854 12.854c-0-0-0-0-0-0l-4.854-4.854 4.854-4.854c0-0 0-0 0-0 0.052-0.052 0.090-0.113 0.114-0.178 0.066-0.178 0.028-0.386-0.114-0.529l-2.293-2.293c-0.143-0.143-0.351-0.181-0.529-0.114-0.065 0.024-0.126 0.062-0.178 0.114 0 0-0 0-0 0l-4.854 4.854-4.854-4.854c-0-0-0-0-0-0-0.052-0.052-0.113-0.090-0.178-0.114-0.178-0.066-0.386-0.029-0.529 0.114l-2.293 2.293c-0.143 0.143-0.181 0.351-0.114 0.529 0.024 0.065 0.062 0.126 0.114 0.178 0 0 0 0 0 0l4.854 4.854-4.854 4.854c-0 0-0 0-0 0-0.052 0.052-0.090 0.113-0.114 0.178-0.066 0.178-0.029 0.386 0.114 0.529l2.293 2.293c0.143 0.143 0.351 0.181 0.529 0.114 0.065-0.024 0.126-0.062 0.178-0.114 0-0 0-0 0-0l4.854-4.854 4.854 4.854c0 0 0 0 0 0 0.052 0.052 0.113 0.090 0.178 0.114 0.178 0.066 0.386 0.029 0.529-0.114l2.293-2.293c0.143-0.143 0.181-0.351 0.114-0.529-0.024-0.065-0.062-0.126-0.114-0.178z"
            />
          </svg>
        </button>
      </header>
      <p />
      <section
        class="rename-section"
      >
        <button
          class="rename-button"
          data-testid="rename-button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="m15 16-4 4h10v-4zM12.06 7.19 3 16.25V20h3.75l9.06-9.06-3.75-3.75zM5.92 18H5v-.92l7.06-7.06.92.92L5.92 18zM18.71 8.04a.996.996 0 0 0 0-1.41l-2.34-2.34a1.001 1.001 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
          board.editColumnPanel.rename
        </button>
      </section>
      <button
        class="remove-button"
        data-testid="remove-button"
      >
        <svg
          color="#b13838"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(177, 56, 56);"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 8H20V21C20 21.5523 19.5523 22 19 22H5C4.44772 22 4 21.5523 4 21V8ZM7 5V3C7 2.44772 7.44772 2 8 2H16C16.5523 2 17 2.44772 17 3V5H22V7H2V5H7ZM9 4V5H15V4H9ZM9 12V18H11V12H9ZM13 12V18H15V12H13Z"
          />
        </svg>
        board.editColumnPanel.remove
      </button>
    </div>
  </div>
</DocumentFragment>
`;
